### ISSUE TITLE
Failed to activate the linter-ui-default package

### ISSUE THREAD

**Issue by ericrowan** [AtomLinter/linter-ui-default#25]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.4.0
**System**: Mac OS X 10.11.2
**Thrown From**: [linter-ui-default](https://github.com/AtomLinter/linter-ui-default) package, v0.0.2

### Stack Trace

Failed to activate the linter-ui-default package

```
At LinterUI is not a function

TypeError: LinterUI is not a function
    at Object.activate (/Users/ericrowan/.atom/packages/linter-ui-default/lib/index.js:44:15)
    at Package.module.exports.Package.activateNow (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/package.js:183:20)
    at /opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/package.js:156:32
    at Package.module.exports.Package.measure (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/package.js:92:15)
    at /opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/package.js:149:26
    at Package.module.exports.Package.activate (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/package.js:146:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/package-manager.js:524:21)
    at /opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:385:60
    at exit (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
    at triggerExitCallback (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:213:47)
    at /opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:220:18
    at Socket.<anonymous> (/opt/homebrew-cask/Caskroom/atom/1.2.4/Atom.app/Contents/Resources/app.asar/src/buffered-process.js:98:18)
    at emitOne (events.js:82:20)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```

### Commands

```
     -0:50.9.0 tree-view:remove-project-folder (span.name.icon.icon-repo)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "dev-live-reload",
      "emmet",
      "pigments",
      "linter-scss-lint",
      "auto-detect-indentation",
      "indent-guide-improved"
    ]
  }
}
```

### Installed Packages

```coffee
# User
autoclose-html, v0.19.0
color-picker, v2.0.14
css-snippets, v0.9.0
language-blade, v0.19.0
language-svg, v0.9.0
laravel-snippets, v2.1.0
linter, v1.11.3
linter-csslint, v1.2.0
linter-handlebars, v2.3.1
linter-js-yaml, v1.2.5
linter-jshint, v1.2.2
linter-jsonlint, v1.1.4
linter-less, v2.3.3
linter-sass-lint, v0.6.3
linter-ui-default, v0.0.2
open-recent, v4.0.1

# Dev
No dev packages
```

**Comment by steelbrain** [comment]
@erow80 That is a very interesting error you hit there, do you have an isolated repro case for this?

**Comment by erow80** [comment]
@steelbrain Not sure what you mean by isolated repo case? But yeah, I hadn't seen an error like this in Atom yet. Not really sure what is going TBH.

**Comment by steelbrain** [comment]
@erow80 Did it happen only once or it happens everytime you restart the package?

**Comment by erow80** [comment]
So far I've seen it only once. I closed Atom and re-opened, and so far so good.

**Comment by steelbrain** [comment]
Fixed by https://github.com/AtomLinter/linter-ui-default/pull/40

**Comment by PixelT** [comment]
@steelbrain 
After update to 1.2.1 I have this error again.
```
Failed to activate the linter-ui-default package
Cannot find module 'sb-react-table'
```

**Comment by steelbrain** [comment]
@PixelT That's actually a rare network issue you hit, do `apm uninstall linter-ui-default` then `apm install linter-ui-default` to fix

**Comment by qk4** [comment]
@steelbrain 
I'm getting a similar error on 1.2.1
```
Failed to activate the linter-ui-default package
(0 , _validate.ui) is not a function
```
It also pops up a 
```
[Linter] Error running jshint
```

I installed linter(+ deps) and jshint and it worked on one file, switched to another and it wasn't working so I closed atom and now get this error when I open it.

Atom linked me here.

**Comment by steelbrain** [comment]
@qk4 Have you tried the solution above? That or just maybe restarting Atom should fix this for you

**Comment by qk4** [comment]
@steelbrain Yeah just tried that, restarted atom and computer. Still seems to load fine but doesn't actually show any errors/warnings/etc

**Comment by steelbrain** [comment]
@qk4 `linter:debug` command is your friend! :)

**Comment by qk4** [comment]
@steelbrain That command didn't show me any obvious problems. However in the dev console it's showing:
```
[Linter] Error running JSHint Error: /home/qk4/.atom/packages/linter-jshint/node_modules/jshint-json/json.js:4
		console.log(JSON.stringify({
		                 ^

TypeError: Converting circular structure to JSON
    at Object.stringify (native)
    at reporter (/home/qk4/.atom/packages/linter-jshint/node_modules/jshint-json/json.js:4:20)
    at Object.<anonymous> (/home/qk4/.atom/packages/linter-jshint/node_modules/jshint/src/cli.js:635:39)
    at Socket.<anonymous> (/home/qk4/.atom/packages/linter-jshint/node_modules/cli/cli.js:844:18)
    at emitNone (events.js:91:20)
    at Socket.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Object.stringify (native)
    at reporter (/home/qk4/.atom/packages/linter-jshint/node_modules/jshint-json/json.js:4:20)
    at Object.<anonymous> (/home/qk4/.atom/packages/linter-jshint/node_modules/jshint/src/cli.js:635:39)
    at Socket.<anonymous> (/home/qk4/.atom/packages/linter-jshint/node_modules/cli/cli.js:844:18)
    at emitNone (events.js:91:20)
    at Socket.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at ChildProcess.<anonymous> (/home/qk4/.atom/packages/linter-jshint/node_modules/sb-exec/lib/index.js:56:20)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:493:12)(anonymous function) @ linter-registry.js [sm]:144
```
I guess I should take this over to https://github.com/AtomLinter/linter-jshint/issues huh?
Thanks for the help.

**Comment by steelbrain** [comment]
Yep, that's the place for it. Nice debugging 👍

**Comment by Swivelgames** [comment]
I'm getting a different error, but `atom` directed me to this issue:

1. Open Atom
2. Receive Prompt to Update linter packages
3. Update linter packages, and restart Atom
4. Receive error

**Atom**: 1.15.0 x64
**Electron**: 1.3.13
**OS**: Mac OS X 10.11.6
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.2.3


### Stack Trace

Failed to activate the linter-ui-default package

```
At _busySignal2.default is not a constructor

TypeError: _busySignal2.default is not a constructor
    at /packages/linter-ui-default/lib/main.js:30:19)
    at Object.provideUI (/packages/linter-ui-default/lib/index.js:36:22)
    at Package.module.exports.Package.activateServices (/app.asar/src/package.js:393:69)
    at Package.module.exports.Package.activateNow (/app.asar/src/package.js:221:16)
    at /app.asar/src/package.js:190:32
    at Package.module.exports.Package.measure (/app.asar/src/package.js:96:15)
    at /app.asar/src/package.js:183:26
    at Package.module.exports.Package.activate (/app.asar/src/package.js:180:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/app.asar/src/package-manager.js:550:34)
    at /app.asar/src/package-manager.js:531:29
    at Config.module.exports.Config.transactAsync (/app.asar/src/config.js:337:18)
    at PackageManager.module.exports.PackageManager.activatePackages (/app.asar/src/package-manager.js:526:19)
    at PackageManager.module.exports.PackageManager.activate (/app.asar/src/package-manager.js:508:46)
    at /app.asar/src/atom-environment.js:812:28
```

### Commands

```
     -0:04.2.0 core:copy (atom-notification.fatal.icon.icon-bug.native-key-bindings.has-detail.has-close.has-stack)
```

### Non-Core Packages

```
atom-alignment 0.12.1 
atom-typescript 11.0.1 
auto-detect-indentation 1.3.0 
busy-signal 1.3.0 
disable-mru-tabbing 0.4.0 
duotone-blue-dark-syntax 1.4.0 
duotone-dark-amethyst-syntax 2.0.4 
duotone-dark-red-syntax 0.3.2 
duotone-dark-sea-syntax 2.1.0 
duotone-dark-space-syntax 2.1.0 
duotone-denim-syntax 0.4.3 
intentions 1.1.2 
language-babel 2.57.3 
linter 2.1.4 
linter-eslint 8.1.6 
linter-ui-default 1.2.3 
minimap 4.27.1 
node-debugger 1.10.1 
pigments 0.39.1 
vim-mode 0.66.0 
```

**Comment by steelbrain** [comment]
@Swivelgames While it's different from the issue posted on, it's one of those good ol' APM bugs. Fix? `apm uninstall busy-signal linter-ui-default` then `apm install busy-signal linter-ui-default`

**Comment by Swivelgames** [comment]
Oddly, the first time I attempted that solution it didn't work. The second time worked, though. Thanks @steelbrain

**Comment by cruzlauroiii** [comment]
[Enter steps to reproduce:]

1. compile atom using git
2. error: using git versions: navigator.appVersion
"5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Atom/1.18.0-dev-1f3bb12a1 Chrome/56.0.2924.87 Electron/1.6.7 Safari/537.36"

**Atom**: 1.18.0-dev-1f3bb12a1 x64
**Electron**: 1.6.7
**OS**: Mac OS X 10.12.5
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.2.3


### Stack Trace

Failed to activate the linter-ui-default package

```
At /Users/stellar/.atom/packages/linter-ui-default/lib/index.js: Buffer is not a constructor

TypeError: /Users/stellar/.atom/packages/linter-ui-default/lib/index.js: Buffer is not a constructor
    at Converter.toBase64 (/Applications/Atom.app/Contents/Resources/app/node_modules/convert-source-map/index.js:61:10)
    at Converter.toComment (/Applications/Atom.app/Contents/Resources/app/node_modules/convert-source-map/index.js:65:21)
    at File.generate (/Applications/Atom.app/Contents/Resources/app/node_modules/babel-core/lib/transformation/file/index.js:812:82)
    at File.transform (/Applications/Atom.app/Contents/Resources/app/node_modules/babel-core/lib/transformation/file/index.js:625:17)
    at /Applications/Atom.app/Contents/Resources/app/node_modules/babel-core/lib/transformation/pipeline.js:168:19
    at File.wrap (/Applications/Atom.app/Contents/Resources/app/node_modules/babel-core/lib/transformation/file/index.js:639:16)
    at Pipeline.transform (/Applications/Atom.app/Contents/Resources/app/node_modules/babel-core/lib/transformation/pipeline.js:165:17)
    at Object.exports.compile (/Applications/Atom.app/Contents/Resources/app/src/babel.js:70:1)
    at Object.compile (/Applications/Atom.app/Contents/Resources/app/src/package-transpilation-registry.js:72:1)
    at compileFileAtPath (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:97:1)
    at Object.value [as .js] (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:229:1)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (Applications/Atom.app/Contents/Resources/app/static/index.js:47:45)
    at require (internal/module.js:20:19)
    at customRequire (/Applications/Atom.app/Contents/Resources/app/static/<embedded>:96:1)
    at Package.module.exports.Package.requireMainModule (/Applications/Atom.app/Contents/Resources/app/src/package.js:887:1)
    at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app/src/package.js:242:1)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:225:1
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:99:1)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:218:1
    at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app/src/package.js:215:1)
    at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:642:1)
    at /Applications/Atom.app/Contents/Resources/app/src/package-manager.js:623:1
    at Config.module.exports.Config.transactAsync (/Applications/Atom.app/Contents/Resources/app/src/config.js:346:1)
    at PackageManager.module.exports.PackageManager.activatePackages (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:618:1)
    at PackageManager.module.exports.PackageManager.activate (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:600:1)
    at /Applications/Atom.app/Contents/Resources/app/src/atom-environment.js:846:1
```

### Commands

```
  3x -5:42.6.0 window:toggle-dev-tools (input.hidden-input)
```

### Non-Core Packages

```
busy-signal 1.3.0 
intentions 1.1.2 
lebab 0.10.0 
linter 2.1.4 
linter-eslint 8.1.6 
linter-ui-default 1.2.3 
remove-whitespace 0.2.1 
symbol-gen 1.3.1 
```

**Comment by philip-hodder** [comment]
I'm getting what the error message says is the same error (hence the post here), although the cause seems different. This occurs whenever I switch projects using the "project-viewer" plugin. I've tried uninstalling and reinstalling "busy-signal" and "linter-ui-default" using apm but no joy.

Thanks!


[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.17.2 x64
**Electron**: 1.3.15
**OS**: Microsoft Windows 7 Enterprise 
**Thrown From**: [linter-ui-default](https://github.com/steelbrain/linter-ui-default) package 1.6.0


### Stack Trace

Failed to activate the linter-ui-default package

```
At Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.

Error: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at Error (native)
    at StatusBarView.module.exports.StatusBarView.addRightTile (~/AppData/Local/atom/app-1.17.2/resources/app/node_modules/status-bar/lib/status-bar-view.js:90:1)
    at /packages/linter-ui-default/lib/status-bar/index.js:91:67
    at Config.module.exports.Config.observeKeyPath (~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:625:1)
    at Config.module.exports.Config.observe (~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:150:1)
    at StatusBar.attach (/packages/linter-ui-default/lib/status-bar/index.js:87:40)
    at /packages/linter-ui-default/lib/index.js:55:26
    at Set.forEach (native)
    at Object.consumeStatusBar (/packages/linter-ui-default/lib/index.js:54:20)
    at Provider.module.exports.Provider.provide (~/AppData/Local/atom/app-1.17.2/resources/app/node_modules/service-hub/lib/provider.js:34:1)
    at ServiceHub.module.exports.ServiceHub.consume (~/AppData/Local/atom/app-1.17.2/resources/app/node_modules/service-hub/lib/service-hub.js:50:1)
    at Package.module.exports.Package.activateServices (~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:454:1)
    at Package.module.exports.Package.activateNow (~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:256:1)
    at ~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:225:1
    at Package.module.exports.Package.measure (~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:99:1)
    at ~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:218:1
    at Package.module.exports.Package.activate (~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:215:1)
    at PackageManager.module.exports.PackageManager.activatePackage (~/AppData/Local/atom/app-1.17.2/resources/app/src/package-manager.js:642:1)
    at ~/AppData/Local/atom/app-1.17.2/resources/app/src/package-manager.js:374:1
    at ~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:644:1
    at Function.module.exports.Emitter.simpleDispatch (~/AppData/Local/atom/app-1.17.2/resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (~/AppData/Local/atom/app-1.17.2/resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at Config.module.exports.Config.emitChangeEvent (~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:835:1)
    at Config.module.exports.Config.setRawValue (~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:621:1)
    at Config.module.exports.Config.set (~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:236:1)
    at Config.module.exports.Config.removeAtKeyPath (~/AppData/Local/atom/app-1.17.2/resources/app/src/config.js:386:1)
    at Package.module.exports.Package.enable (~/AppData/Local/atom/app-1.17.2/resources/app/src/package.js:84:1)
    at PackageManager.module.exports.PackageManager.enablePackage (~/AppData/Local/atom/app-1.17.2/resources/app/src/package-manager.js:185:1)
    at list.forEach.pkg (/packages/project-viewer/src/packages.js:16:42)
    at Array.forEach (native)
    at /packages/project-viewer/src/packages.js:14:8)
    at /packages/project-viewer/src/project-view.js:303:22)
    at /packages/project-viewer/src/project-view.js:17:8)
```

### Commands

```
  2x -2:00.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always)
```

### Non-Core Packages

```
activate-power-mode 2.0.0 
advanced-open-file 0.16.6 
api-docs 0.0.6 
ask-stack 2.2.0 
atom-beautify 0.29.26 
atom-html-preview 0.1.22 
atom-math 0.0.9 
atom-ungit 0.5.3 
atom-webbrowser 2.1.4 
atom-wrap-in-tag 0.6.0 
atomatigit 1.5.5 
autoclose-html 0.23.0 
autocomplete-date 0.4.0 
autocomplete-paths 1.0.5 
autocomplete-xml 0.9.4 
block-comment-plus 0.4.0 
build-tools 4.5.9 
build-tools-ant 0.1.1 
busy-signal 1.4.1 
character-table 0.4.3 
charcode-display 0.2.1 
color-picker 2.2.5 
custom-title 1.0.1 
dark-flat-ui 1.2.4 
dash 1.7.1 
document-outline 1.5.0 
dumb-quoter 0.1.0 
expose 0.14.0 
faker-atom 1.0.0 
fark-syntax 0.1.0 
file-icons 2.1.7 
filesize 3.0.0 
flex-tool-bar 0.12.0 
fonts 2.1.0 
git-plus 7.9.2 
git-time-machine 1.5.9 
hey-pane 1.0.0 
highlight-selected 0.13.1 
hybrid-reduced-contrast-syntax 0.0.1 
imdone-atom 2.2.6 
intentions 1.1.2 
isotope-ui 2.8.5 
language-diff 0.7.0 
language-jsoniq 1.8.0 
language-markdown 0.22.0 
language-pfm 0.82.0 
lines 0.13.1 
linter 2.1.4 
linter-autocomplete-jing undefined 
linter-htmlhint 1.3.3 
linter-js-cloudformation-yaml 2.2.1 
linter-jshint 3.1.2 
linter-ui-default 1.6.0 
linter-write-good 0.9.0 
markdown-document 0.2.1 
markdown-image-assistant 0.1.8 
markdown-preview-plus 2.4.9 
markdown-table-editor 0.5.2 
merge-conflicts 1.4.5 
minimap 4.28.2 
minimap-highlight-selected 4.6.1 
minimap-linter 2.0.0 
minimap-pigments 0.2.2 
minimap-split-diff 0.3.7 
minimap-titles 1.11.0 
native-ui 0.20.6 
nav-panel 0.0.11 
open-recent 5.0.0 
pdf-view 0.58.0 
pen-paper-coffee-syntax 0.14.9 
pigments 0.39.1 
pinned-tabs 1.1.0 
preview-inline 1.4.7 
project-viewer 1.1.2 
random 1.3.0 
smart-quotes-plus 2.1.0 
solarized-dark-ui 0.3.2 
split-diff 1.4.0 
Sublime-Style-Column-Selection 1.7.4 
svg-preview 0.11.0 
sync-settings 0.8.1 
tablr 1.8.3 
tag 0.5.0 
text-manipulation 0.6.0 
theme-switcher 1.1.0 
tidy-markdown 3.0.1 
todo-show 1.11.0 
tool-bar 1.1.0 
tree-view-git-branch 0.1.2 
tree-view-git-status 1.4.0 
update-changelogs 0.1.0 
wordcount 2.10.4 
working-files 0.6.5 
xml-common-schemata 0.0.3 
Zen 0.18.0 
zentabs 0.8.8 
```

**Comment by Arcanemagus** [comment]
@cruzlauroiii I know that later parts in the tree don't have the issue you posted, so I'm just chalking that up to a bug in Atom's `master` that was fixed. If you can reproduce on the current Stable or Beta builds please let us know!

@philip-hodder Can you please file that as a new issue so it can be tracked down? Thanks!

Locking this thread, if you are directed here by Atom as an "existing issue" please file a new issue! Atom groups bugs together by title, not the stack trace so many different bugs get lumped together which is problematic for fixing them.
